.github/workflows/pip-compile.yml

```diff
@@ -0,0 +1,77 @@
+---
+name: "Refresh pinned dependencies"
+
+"on":
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+  push:
+    branches:
+      - devel
+    paths:
+      - .github/workflows/pip-compile.yml
+      - "tests/*.in"
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  refresh:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: "devel"
+      - name: Fetch required contents of ansible-core
+        run: |
+          python docs/bin/clone-core.py
+      - name: Set up nox
+        uses: wntrblm/nox@2023.04.22
+        with:
+          python-versions: "3.9"
+      - name: Use a branch named pip-compile
+        id: branch
+        run: |
+          git config user.name "Github Actions"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+          if git branch -r | grep origin/pip-compile; then
+            echo "branch-exists=true" >> "${GITHUB_OUTPUT}"
+            git switch pip-compile
+            git rebase devel
+          else
+            echo "branch-exists=false" >> "${GITHUB_OUTPUT}"
+            git switch -c pip-compile
+          fi
+      - name: "Run nox -e pip-compile"
+        env:
+          # Ensure the latest pip version is used
+          VIRTUALENV_DOWNLOAD: '1'
+        run: |
+          nox -e pip-compile
+      - name: Push new dependency versions and create a PR
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```

Comment:
This will likely cause PRs with no CI triggered, unless I'm overlooking something. Just something to look out for. But that'll become visible post-merge.

Reply:
Why? It worked when I tested it locally.

Reply:
You can't test locally with a special-cased token that GHA injects — this token's side effect caused events don't propagate across GitHub's event platform and no integrations receive notifications through webhooks. In practice, this means that no GitHub Apps or GHA will be triggered on pushes made with this token.
You'll see what I mean after merge, I suppose.

Reply:
It WFM when I tested it in my fork repo, but I guess we'll see.